src/net/p2p/managerbase.h

```diff
@@ -142,6 +143,19 @@ namespace P2P {
        */
       void trySpawnInboundSession(tcp::socket&& socket);
 
+      /**
+       * Insert a new request into requests_.
+       * @param request The newly created request to be tracked by requests_.
+       */
+      void insertRequest(const std::shared_ptr<Request>& request);
```

Comment:
Since this function intends to take ownership of the `std::shared_ptr` copy, it would be a little bit better to receive it by value instead of const-ref.

Like that:
```c++
void insertRequest(std::shared_ptr<Request> request) {
  requests_[activeRequests_][request->id()] = std::move(request); // note the move
  // ...
}
```

Moving a `shared_ptr` is cheaper than copying it. At least in this way, you delegate to the caller the decision of copying or moving it (instead of always copying).

I'd suggest also making such change in other places too, where a `std::shared_ptr` parameter is received and will be stored somewhere, but such change could fall out of the scope of this PR. **No problem at all if you decide not to do it**.